CODING_GUIDELINES.md

```diff
@@ -211,6 +211,62 @@ a += "b"
   + "d";
 ```
 
+* Initializers
+
+We allow brace enclosed initializers only for aggregates and arrays/containers.<br />
+Brace enclosed initializer MUST be used with equality sign if it follow the variable declaration.<br />
+Brace enclosed initializer MUST be additionally enclosed in parentheses if it used in constructor initialization list.<br />
+Some valid use cases:
+```c++
+// aggregate
+Person john = { "John", "Smith", 21 };
+Person *john = new Person { "John", "Smith", 21 };
+
+// array
+int array[] = { 1, 2, 3, 4 };
+
+// container
+QHash<QString, QString> map = {
+    { "key1", "value1" },
+    { "key2", "value2" }
+);
+
+// member array
+SomeClass::SomeClass(BaseClass *parent)
+    : BaseClass(parent)
+    , m_someArrayMember({ 1, 2, 3, 4 })
+{
+}
+
+// return from function
+Person getPersonByName(const QString &name)
+{
+    // do something
+    return { name, surname, age };
+}
+
+// function argument
+doSomething({ name, surname, age }, someOtherArg);
+```
+
+* **auto** keyword
+
+We allow to use the **auto** keyword only where it doesn't break the readability of the code (i.e. either we can gather enough information about the type from the right part of the expression, or we do not need to know the exact type), or where it is strictly necessary (for example, to compute the type of a lambda, etc.).<br />
+Some valid use cases:
+```c++
+foreach (auto item, container) {
```

Comment:
You can ignore this: Personally I like to use the concrete type in order to specify "const" where it can be specified. This use case probably falls under the category of "premature optimization". I just hope users don't think that this rule enforces the auto keyword in foreach.